### PR TITLE
chore(build): bump nfpm to v2.31.0 + use tree nfpm type

### PR DESCRIPTION
**Pull request by gszr**
The `tree` type is a new feature of nfpm that we can leverage to improve
the packaging of artifacts.

Use `tree` type in nfpm config so that the directory tree is replicated
in the destination, preserving the symlinks. With the default `file`
type, only symlinks to regular files are preserved, while symlinks to
directories are not.

### CODE REVIEW

File: build/dockerfiles/deb.Dockerfile [hunk 1]
```diff
@@ -27,9 +27,6 @@ RUN apt-get update \
     && ln -sf /usr/local/openresty/luajit/bin/luajit /usr/local/bin/luajit \
     && ln -sf /usr/local/openresty/luajit/bin/luajit /usr/local/bin/lua \
     && ln -sf /usr/local/openresty/nginx/sbin/nginx /usr/local/bin/nginx \
-    && if [ -d /usr/local/openresty/nginx/modules ]; then \
-        ln -sf /usr/local/openresty/nginx/modules /usr/local/kong/modules ; \
-       fi; \
     kong version
```
Comment (locao) [inline build/dockerfiles/deb.Dockerfile:30]:
```suggestion
    && kong version
```

File: build/dockerfiles/rpm.Dockerfile [hunk 2]
```diff
@@ -40,9 +40,6 @@ RUN yum install -y /tmp/kong.rpm \
     && ln -sf /usr/local/openresty/luajit/bin/luajit /usr/local/bin/luajit \
     && ln -sf /usr/local/openresty/luajit/bin/luajit /usr/local/bin/lua \
     && ln -sf /usr/local/openresty/nginx/sbin/nginx /usr/local/bin/nginx \
-    && if [ -d /usr/local/openresty/nginx/modules ]; then \
-        ln -sf /usr/local/openresty/nginx/modules /usr/local/kong/modules ; \
-       fi; \
     kong version
```
Comment (locao) [inline build/dockerfiles/rpm.Dockerfile:43]:
```suggestion
    && kong version
```

**Review by locao — CHANGES_REQUESTED** [review]
Simple fix

**Comment by gszr** [comment]
NOTE: release jobs will be failing for now - til we rebase onto latest master to include Cloudsmith work.

**Review by locao — APPROVED** [review]

**Pull request merged**